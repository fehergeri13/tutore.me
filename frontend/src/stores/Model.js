import {observable, action} from "mobx";

import Auth from "./Auth";
import Rest from "./Rest";

export default class Model {
    @observable.ref auth = new Auth(this);
    @observable.ref routingStore = undefined;
    @observable.ref rest = new Rest(this);

    @observable isMessageModalOpen = false;
    @observable messageUserId = undefined;
    @observable messageUsername = undefined;


    constructor(routingStore) {
        this.routingStore = routingStore;
    }

    @action openMessageModal(messageUserId, messageUsername) {
        this.isMessageModalOpen = true;
        this.messageUserId = messageUserId;
        this.messageUsername = messageUsername;
    }

    @action closeMessageModal() {
        this.isMessageModalOpen = false;
    }
}