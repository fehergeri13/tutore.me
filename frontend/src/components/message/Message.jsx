import React from 'react';
import {observable, toJS} from 'mobx';
import {inject, observer} from 'mobx-react';
import autobind from 'autobind-decorator';
import "./message.less";
import Modal from 'react-modal';

@inject('model')
@observer
export default class Message extends React.Component {

    @observable.shallow data = {
        message: "",
    };

    @autobind
    handleChange(e) {
        const target = e.target;
        const value = target.type === 'checkbox' ? target.checked : target.value;
        const name = target.name;

        this.data[name] = value;

        console.log('loginData', toJS(this.data))
    }

    @autobind
    handleSubmit(e) {
        e.preventDefault();

        this.props.model.rest.sendMessage(this.props.model.messagePost.userId, this.data.message);
    }

    @autobind
    handleCancelClick(e) {
        e.preventDefault();

        this.props.model.closeMessageModal();
    }

    render() {
        if(!this.props.model.isMessageModalOpen) {
            return null;
        }

        return <Modal
            isOpen={this.props.model.isMessageModalOpen}
            className={{
                base: 'message-modal',
                afterOpen: 'myClass_after-open',
                beforeClose: 'myClass_before-close',
            }}
            overlayClassName={{
                base: 'message-modal-overlay',
                afterOpen: 'myOverlayClass_after-open',
                beforeClose: 'myOverlayClass_before-close',
            }}
        >
            <h2>Üzenet küldése</h2>

            <form onSubmit={this.handleSubmit}>
                <input
                    type="text"
                    value={this.props.model.messagePost.username}
                    readOnly
                    />
                <input
                    type="text"
                    name="message"
                    onChange={this.handleChange}
                    value={this.data.message}
                />
                <button
                type="submit">
                    Küld
                </button>
                <button
                    type="button"
                    onClick={this.handleCancelClick}
                >Mégsem küldök üzenetet</button>
            </form>
        </Modal>
    }
}
