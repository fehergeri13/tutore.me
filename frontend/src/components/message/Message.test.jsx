import React from 'react';
import renderer from 'react-test-renderer';
import Message from "./Message";
import {MemoryRouter as Router} from "react-router-dom";
import {Provider} from 'mobx-react'
import Model from "../../stores/Model";

describe('PostItem', () => {
    it('renders correctly', () => {

        const model = new Model();

        const tree = renderer.create(
            <Provider model={model}><Router><Message /></Router></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
