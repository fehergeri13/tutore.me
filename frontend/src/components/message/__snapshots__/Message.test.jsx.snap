// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem renders correctly 1`] = `null`;
